discovery/gossiper.go

```diff
@@ -2043,21 +2043,13 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 				remotePub, _ = chanInfo.NodeKey1()
 			}
 
-			sPeer, err := d.cfg.FindPeer(remotePub)
-			if err != nil {
-				log.Errorf("unable to send channel update -- "+
-					"could not find peer %x: %v",
-					remotePub, err)
-			} else {
-				// Send ChannelUpdate directly to remotePeer.
-				// TODO(halseth): make reliable send?
-				err = sPeer.SendMessage(false, msg)
-				if err != nil {
-					log.Errorf("unable to send channel "+
-						"update message to peer %x: %v",
-						remotePub.SerializeCompressed(),
-						err)
-				}
+			// Now, we'll attempt to send the channel update message
+			// reliably to the remote peer in the background, so
+			// that we don't block if the peer happens to be offline
+			// at the moment.
+			if err := d.sendMsgReliably(msg, remotePub); err != nil {
+				nMsg.err <- err
+				return nil
```

Comment:
This alters existing bahaviour. Intentional?

Reply:
I think it makes sense to return the error here, to be clear.

Reply:
Yeah this was intentional, I also think it makes sense to return the error here.